src/main/java/com/bp3/camunda/camunda7/C7RESTConnector.java

```diff
@@ -79,17 +84,57 @@ public void execute(ExternalTask externalTask, ExternalTaskService externalTaskS
         setPayload(request, httpMethod, httpPayload);
 
         // call the REST service
-        HttpResponse response = request.execute();
+        try {
+            HttpResponse response = request.execute();
+
+            // set the output variable
+            log.debug("RESPONSE: {}", response.getResponse());
+            VariableMap variables = Variables.createVariables();
+            if (outputVariableName != null) {
+                variables.putValue(outputVariableName, response.getResponse());
+            }
 
-        // set the output variable
-        log.debug("RESPONSE: {}", response.getResponse());
-        VariableMap variables = Variables.createVariables();
-        if (outputVariableName != null) {
-            variables.putValue(outputVariableName, response.getResponse());
-        }
+            // complete the external task
+            externalTaskService.complete(externalTask, variables);
+        } catch(ConnectorException e) {
+            log.error("CONNECTOR_ERROR: {}", e.getLocalizedMessage(), e.fillInStackTrace());
```

Comment:
Are you trying to hide part of the stack trace? Is that why you are calling `fillInStackTrace`?

Reply:
To be fair `fillInStackTrace` was unnecessary, so changed to pass in the root exception .

Reply:
Ok, that's what I was really wondering, was it necessary. Thanks for checking it and updating.